fetcher/fetcher.go

```diff
@@ -50,11 +41,12 @@ type BulkResponse struct {
 
 // Fetcher is a struct responsible for fetching nodes endpoints
 type Fetcher struct {
-	file         *os.File
-	client       *http.Client
-	endpoints    <-chan EndpointRequest
-	statusUpdate chan<- StatusUpdate
-	results      chan<- BulkResponse
+	file          *os.File
+	client        *http.Client
+	endpoints     <-chan EndpointRequest
+	statusUpdate  chan<- StatusUpdate
+	results       chan<- BulkResponse
+	prometheusVec prometheus.ObserverVec
```

Comment:
I think the name could be more descriptive and/or type could be more precise. What exactly we will collect there?